src/client/api/init.c

```diff
@@ -143,6 +146,23 @@ daos_init(void)
 	if (rc != 0)
 		D_GOTO(unlock, rc);
 
+	d_fault_init = d_fault_attr_lookup(101);
+
+	/* If fault injection 101 is set then turn off fault injection 0 for the rest of this
+	 * function.  This allows us to only test daos_init() under fault injection for one
+	 * test only, then avoid replicating the same effort for other fault injection tests.
+	 */
+	if (D_SHOULD_FAIL(d_fault_init)) {
+		struct d_fault_attr_t blank = {};
+
+		d_fault_mem = d_fault_attr_lookup(0);
+
+		if (d_fault_mem) {
+			d_fault_mem_saved = *d_fault_mem;
+			d_fault_attr_set(0, blank);
+		}
```

Comment:
Couldn't we just do this?
```suggestion
		d_fault_mem_saved = d_fault_attr_lookup(0);

		if (d_fault_mem_saved)
			d_fault_attr_set(0, blank);
```

Reply:
No, because d_fault_attr_lookup() returns you a pointer to the settings, so then when you call d_fault_attr_set() then you overwrite them, and the later d_fault_attr_set would not re-enable anything.

Reply:
Ah.... that's a little strange :)

Reply:
So is the massive "pass by value" struct in the API.  @frostedcmos any comments?